File: tests/machine/x/pascal/record_assign.pas
program RecordAssign;
{$mode objfpc}
{$modeswitch nestedprocvars}

uses SysUtils, fgl, fphttpclient, Classes, Variants, fpjson, jsonparser;

type
  generic TArray<T> = array of T;

type Counter = record
  n: integer;
end;

function inc(c: Counter): integer;
begin
  c := c.n + 1;
end;

var
  _tmp0: Counter;
  c: Counter;

begin
  _tmp0.n := 0;
  c := _tmp0;
  inc(c);
  writeln(c.n);
end.

File: tests/machine/x/cs/record_assign.cs
using System;

public record struct Counter {
    public int n;
}

class Program {
    static void inc(Counter c) {
        c.n = (c.n + 1);
    }
    
    static void Main() {
        Counter c = new Counter { n = 0 };
        inc(c);
        Console.WriteLine(c.n);
    }
}
